src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -123,9 +123,19 @@ class CompilerContainer {
       if (pragmaArr && pragmaArr.length === 1) {
         const pragmaStr = pragmaArr[0].replace('pragma solidity', '').trim()
         const pragma = pragmaStr.substring(0, pragmaStr.length - 1)
-        const fixedVersions = this.data.allversions.filter(obj => !obj.prerelease).map(obj => obj.version)
-        if (!semver.satisfies(this._retrieveVersion(), pragma)) {
-          const compilerToLoad = semver.maxSatisfying(fixedVersions, pragma)
+        const releasedVersions = this.data.allversions.filter(obj => !obj.prerelease).map(obj => obj.version)
+        const allVersions = this.data.allversions.map(obj => this._retrieveVersion(obj.version))
+        const currentCompilerName = this._retrieveVersion(this._view.versionSelector.selectedOptions[0].label)
+        // contains only numbers part, for example '0.4.22'
+        const pureVersion = this._retrieveVersion()
+        // is nightly build newer than the last release
+        const isNewestNightly = currentCompilerName.includes('nightly') && semver.gt(pureVersion, releasedVersions[0])
+        // checking if the selected version is in the pragma range
+        const isInRange = semver.satisfies(pureVersion, pragma)
+        // checking if the selected version is from official compilers list(excluding custom versions) and in range or greater
+        const isOfficial = allVersions.includes(currentCompilerName)
+        if (isOfficial && (!isInRange && !isNewestNightly)) {
```

Comment:
will this be working fine when we don't load custom compiler as this condition fails if compiler version is in pragma range. Let me know if I am missing anything

Reply:
but if the version is in range why do we need to change it?